src/Behavioral.Automation/Services/ComplexBindingBuilder.cs

```diff
@@ -0,0 +1,131 @@
+        }
+
+        public void BuildAction<T>(Action<T> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        public void BuildAction<T1, T2>(Action<T1, T2> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        public void BuildAction<T1, T2, T3>(Action<T1, T2, T3> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        public void BuildAction<T1, T2, T3, T4>(Action<T1, T2, T3, T4> method, params object[] pars)
+        {
+            IEnumerable<StepDefinitionBaseAttribute> attributes = method.GetMethodInfo().GetCustomAttributes<StepDefinitionBaseAttribute>(true);
+            BuildAction(pars, attributes, method.GetMethodInfo().Name);
+        }
+
+        private void BuildAction(object[] pars, IEnumerable<StepDefinitionBaseAttribute> attributes, string methodName)
+        {
+            string definitionType = _scenarioContext.StepContext.StepInfo.StepDefinitionType.ToString();
+            var attributeForCurrentStep = attributes.FirstOrDefault(a => a.GetType().Name.StartsWith(definitionType, StringComparison.OrdinalIgnoreCase));
+            if (attributeForCurrentStep == null)
+            {
+                NUnit.Framework.Assert.Inconclusive($"{definitionType} attribute is not provided for {methodName} binding method");
+            }
+
+            string stepExpression;
+            var table = pars.FirstOrDefault(d => d.GetType() == typeof(Table));
+            if (table != null)
+            {
+                var parsList = pars.ToList();
+                parsList.Remove(table);
+                stepExpression = _stepParametersProcessor.CreateStepExpression(attributeForCurrentStep.Regex, parsList.ToArray());
+                RunAction(stepExpression, (Table)table);
+            }
+            else
+            {
+                stepExpression = _stepParametersProcessor.CreateStepExpression(attributeForCurrentStep.Regex, pars);
+                RunAction(stepExpression);
+            }
+            TestContext.WriteLine($"{new String(' ', _indendationLevel * _indentationSize)}{stepExpression}");
+        }
+
+        private void RunAction(string stepExpression)
+        {
+            switch (_scenarioContext.StepContext.StepInfo.StepDefinitionType)
+            {
+                case TechTalk.SpecFlow.Bindings.StepDefinitionType.Given:
```

Comment:
Can we move full type path to `using` directive?

Reply:
Moved